librz/include/rz_core.h

```diff
@@ -1056,12 +1056,24 @@ RZ_API bool rz_core_meta_string_add(RzCore *core, ut64 addr, ut64 size, RzStrEnc
 RZ_API bool rz_core_meta_pascal_string_add(RzCore *core, ut64 addr, RzStrEnc encoding, RZ_NULLABLE const char *name);
 
 // cprint.c
+typedef enum {
+	RZ_CORE_PRINT_FORMAT_TYPE_OCTAL = 0,
+	RZ_CORE_PRINT_FORMAT_TYPE_INTEGER,
+	RZ_CORE_PRINT_FORMAT_TYPE_HEXADECIMAL,
+	RZ_CORE_PRINT_FORMAT_TYPE_INVALID,
+} RzCorePrintFormatType;
+
 RZ_API RZ_OWN char *rz_core_print_string_c_cpp(RzCore *core);
 RZ_API RZ_OWN char *rz_core_hex_of_assembly(RzCore *core, const char *assembly);
 RZ_API RZ_OWN char *rz_core_esil_of_assembly(RzCore *core, const char *assembly);
 RZ_API RZ_OWN char *rz_core_assembly_of_hex(RzCore *core, ut8 *hex, int len);
 RZ_API RZ_OWN char *rz_core_esil_of_hex(RzCore *core, ut8 *hex, int len);
 
+RZ_API RZ_OWN char *rz_core_print_hexdump_diff_str(RZ_NONNULL RzCore *core, ut64 aa, ut64 ba, ut64 len);
+RZ_API RZ_OWN char *rz_core_print_dump_str(RZ_NONNULL RzCore *core, RzOutputMode mode, ut64 addr, ut8 n, int len, RzCorePrintFormatType format);
+RZ_API RZ_OWN char *rz_core_print_hexdump_or_hexdiff_str(RZ_NONNULL RzCore *core, RzOutputMode mode, ut64 addr, int len, bool use_comment);
+RZ_API RZ_OWN char *rz_core_print_hexdump_byline_str(RZ_NONNULL RzCore *core, bool hex_offset, ut64 addr, int len, ut8 size);
+
```

Comment:
IIRC we said these functions should actually print and not be called `_str` (probably `_print`). Cutter or others can use internal functions like `rz_print_hexdiff_str` and similar, right?

Reply:
@ret2libc if they print they should not be an API, if they should be an API they should return a string. Having an API returning nothing and just printing to the screen is absolutely useless from the user point of view.

Reply:
I moved these to `librz/core/core_private.h`

Reply:
@XVilka we use those kind of rzcore APIs in cases where we need to print the same things in different places (e.g. `rz_core_asm_plugins_print`). But yes, if we think it's not needed right now, let's not make them APIs.